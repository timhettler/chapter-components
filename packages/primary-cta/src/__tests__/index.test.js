import React from 'react';
import Block from '../';
import renderer from 'react-test-renderer';

it(`${Block.name} renders correctly as a button`, () => {
  const tree = renderer.create(<Block label="FOO" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`${Block.name} renders correctly as a link`, () => {
  const tree = renderer
    .create(<Block label="FOO" baseElement="a" href="http://www.rga.com/" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
