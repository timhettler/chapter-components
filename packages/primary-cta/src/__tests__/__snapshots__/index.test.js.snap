// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryCta renders correctly as a button 1`] = `
<div>
  <button
    aria-label="FOO"
    className="primaryCta is-dark"
  >
    <div
      className="outline outline-a"
    />
    <div
      className="outline outline-b"
    />
    <div
      className="text text-a"
    >
      FOO
    </div>
    <div
      aria-hidden="true"
      className="text text-b"
    >
      FOO
    </div>
  </button>
</div>
`;

exports[`PrimaryCta renders correctly as a link 1`] = `
<div>
  <a
    aria-label="FOO"
    className="primaryCta is-dark"
    href="http://www.rga.com/"
  >
    <div
      className="outline outline-a"
    />
    <div
      className="outline outline-b"
    />
    <div
      className="text text-a"
    >
      FOO
    </div>
    <div
      aria-hidden="true"
      className="text text-b"
    >
      FOO
    </div>
  </a>
</div>
`;
